xchange-coinbasepro/src/main/java/org/knowm/xchange/coinbasepro/CoinbaseProAdapters.java

```diff
@@ -255,35 +238,37 @@ public static Order adaptOrder(CoinbaseProOrder order) {
 
   /** The status from the CoinbaseProOrder object converted to xchange status */
   public static OrderStatus adaptOrderStatus(CoinbaseProOrder order) {
-
-    if (order.getStatus().equals("pending")) {
-      return OrderStatus.PENDING_NEW;
+    if (order.getStatus() == null) {
+      return OrderStatus.UNKNOWN;
     }
 
-    if (order.getStatus().equals("done") || order.getStatus().equals("settled")) {
-
-      if (order.getDoneReason().equals("filled")) {
-        return OrderStatus.FILLED;
-      }
-
-      if (order.getDoneReason().equals("canceled")) {
-        return OrderStatus.CANCELED;
-      }
-
-      return OrderStatus.UNKNOWN;
+    switch (order.getStatus()) {
+      case "pending":
+        return OrderStatus.PENDING_NEW;
+      case "done":
+      case "settled":
+        if (order.getDoneReason() == null) {
+          return OrderStatus.UNKNOWN;
+        }
+        switch (order.getDoneReason()) {
+          case "filled":
+            return OrderStatus.FILLED;
+          case "canceled":
+            return OrderStatus.CANCELED;
+        }
+        return OrderStatus.UNKNOWN;
     }
 
     if (order.getFilledSize().signum() == 0) {
 
-      if (order.getStatus().equals("open") && order.getStop() != null) {
+      if ("open".equals(order.getStatus()) && order.getStop() != null) {
```

Comment:
I don't see why we are stating to prefer this type of Yoda conditions?

it looks strange like this would: `null != order.getStop()`

Reply:
I am not following, what do you mean, what is a yoda condition?

Reply:
doing `"open".equals(order.getStatus())` instead of `order.getStatus().equals("open")`

https://en.wikipedia.org/wiki/Yoda_conditions

Reply:
Ah! that is funny I have never heard the term yoda condition. In this case we have added protection, if order.getStatus() returned null, we would do null.equals which would NPE, however by doing constant left side we will not NPE, the way I have written the function is effectively equivilant to order.getStatus() != null && order.getStatus().equals("open)

Reply:
if `order.getStatus()` will return null then the earlier `switch (order.getStatus()) {` will throw an NPE either way

however we can leave it as is if its not confusing to any one else